timer.R

```diff
@@ -0,0 +1,29 @@
+
+library(tictoc)
+
+# Function to source a script and measure its execution time
+time_script <- function(script_path) {
+  tic(paste("Timing", script_path))
+  source(script_path)
+  toc()
+}
+
+# Define the paths to the scripts
+script_paths <- c(
+  "Scripts/method1.r",
+  "Scripts/method2.r",
+  "Scripts/method3.r"
+)
+
+# Run and time each script
+lapply(script_paths, time_script)
+
+
+# Analysis of Execution Times:
+
+# - Original script (method1): 65.75 seconds
+# - Parallel loop script (method2): 46.523 seconds
+# - Parallel function script (method3): 39.116 seconds
+
+# Script 3 is the fastet of the three. This efficiency is due to mroe granular parallelism, allowing
+# for better CPU utilization during the dcomputationally intesive simulations. 
```

Comment:
The reason that the last one is quicker than the second one is because the second one parallelizes the rows in df. Because there are only 5 rows, having more than 5 cores will not make it quicker. The last one parallizes M, and can therefore utilize more cores in the computation. For example, if you use 10 cores, each core will run M/10 simulations each.